### ISSUE TITLE
`wasmi_wasi` cannot be built on Windows on Rust `nightly`

### ISSUE THREAD

**Issue by Berrysoft** [paritytech/wasmi#564]
Thanks for #563 solving the problem, but a degraded `wiggle` introduces an old `cap-primitives`, which introduces more problems:

```
error[E0277]: the trait bound `file_type::FileType: std::sealed::Sealed` is not satisfied                                                                                                 
   --> C:\Users\Straw\.cargo\registry\src\github.com-1ecc6299db9ec823\cap-primitives-0.25.3\src\fs\file_type.rs:134:6
    |
134 | impl std::os::windows::fs::FileTypeExt for FileType {
    |      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ the trait `std::sealed::Sealed` is not implemented for `file_type::FileType`
    |
    = help: the following other types implement trait `std::sealed::Sealed`:
              Child
              Command
              ExitCode
              ExitStatus
              ExitStatusError
              File
              OsStr
              OsString
            and 11 others
note: required by a bound in `std::os::windows::fs::FileTypeExt`
   --> D:\Straw\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib/rustlib/src/rust\library\std\src\os\windows\fs.rs:507:24
    |
507 | pub trait FileTypeExt: Sealed {
    |                        ^^^^^^ required by this bound in `std::os::windows::fs::FileTypeExt`

For more information about this error, try `rustc --explain E0277`.
error: could not compile `cap-primitives` due to previous error
```

**Comment by Robbepop** [comment]
This WASI support is slowly turning into a can of worms ... :upside_down_face:

**Comment by Robbepop** [comment]
Can you give instructions on how to replicate this? I am a bit confused since our GitHub Actions CI did not report this Windows build issue even though we test our workspace on Linux, MacOS and Windows.

**Comment by Berrysoft** [comment]
It seems like a change on rust 1.65.

https://github.com/rust-lang/rust/commit/a4cb0b90c049fd8bc6a4a13edad69d34b98627f0

The `FileTypeExt` is marked as `Sealed` to make 3rd-party types cannot impl it.

**Comment by Robbepop** [comment]
Thanks for the info. Still strange, I was under the impression that our CI was using the current stable versions of Rust which should be 1.65 at the moment.

**Comment by Berrysoft** [comment]
Well, currently I'm using nightly, but as the commit is marked as 1.65, I think the stable should report the same error...

**Comment by Robbepop** [comment]
Ah look what I found:

Windows (Sealed): https://doc.rust-lang.org/std/os/windows/fs/trait.FileTypeExt.html
WASI (not Sealed): https://doc.rust-lang.org/std/os/wasi/fs/trait.FileTypeExt.html

So maybe you need to compile for WASI target which makes sense imo.

**Comment by Robbepop** [comment]
> Well, currently I'm using nightly, but as the commit is marked as 1.65, I think the stable should report the same error...

Can you try using `stable` and see if the problem still persists?

**Comment by Berrysoft** [comment]
> Can you try using stable and see if the problem still persists?

I see. The stable toolchain compiles with no error. Maybe it is still a nightly-specific issue.

**Comment by Berrysoft** [comment]
This issue *has* been solved in recent versions, but I don't think they will fix for an old version:)

**Comment by Robbepop** [comment]
> This issue _has_ been solved in `cap-primitives` 1.0, but `wiggle` 1.0 depends on `cap-primitive` 0.25. Now the Wasmtime team is developing `wiggle` 2.x, and I don't think they will fix this issue for 1.0 releases.

One could this ask nicely. :blush:
